pkg/apis/agones/v1/gameserver.go

```diff
@@ -417,6 +423,22 @@ func (gss *GameServerSpec) Validate(devAddress string) ([]metav1.StatusCause, bo
 					Message: ErrHostPortDynamic,
 				})
 			}
+
+			if p.ContainerName != "" && gss.Container != "" {
+				containerFound := false
+				for _, container := range gss.Template.Spec.Containers {
```

Comment:
Would it make more sense to update `FindGameServerContainer` to something that can `FindContainer` which takes a container name passed in?

Might make this code a little nicer, and likely easier to test.